From 899099be7f522d59baa51d26077c0f5804161cfe Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@linaro.org>
Date: Fri, 29 May 2020 15:27:06 +0200
Subject: [PATCH] xtest: regression 1027/1028: skip tests when test level is 0

Skip regression tests 1027 and 1028 when xtest runs with test level 0
since these tests expect changes in Linux kernel that are not available
in mainline kernel source trees, at least up to v5.7.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
---
 host/xtest/regression_1000.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/host/xtest/regression_1000.c b/host/xtest/regression_1000.c
index a70a722..2b0140c 100644
--- a/host/xtest/regression_1000.c
+++ b/host/xtest/regression_1000.c
@@ -2096,6 +2096,12 @@ static void xtest_tee_test_1027(ADBG_Case_t *c)
 	TEEC_UUID uuid_ns = { };
 	char uuid_name[TEE_UUID_NS_NAME_SIZE] = { };
 
+	/* Skip test at level 0 since its dependency on patched Linux kernel */
+	if (level == 0) {
+		Do_ADBG_Log("Test level 0: skip regression 1027");
+		return;
+	}
+
 	result = xtest_uuid_from_str(&uuid_ns, client_uuid_linux_ns);
 
 	if (!ADBG_EXPECT_TEEC_SUCCESS(c, result))
@@ -2159,6 +2165,12 @@ static void xtest_tee_test_1028(ADBG_Case_t *c)
 	char uuid_name[TEE_UUID_NS_NAME_SIZE] = { };
 	uint32_t group = 0;
 
+	/* Skip test at level 0 since its dependency on patched Linux kernel */
+	if (level == 0) {
+		Do_ADBG_Log("Test level 0: skip regression 1027");
+		return;
+	}
+
 	group = getegid();
 
 	result = xtest_uuid_from_str(&uuid_ns, client_uuid_linux_ns);
-- 
2.17.1

